contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -516,50 +572,56 @@ def getDeployment(cls, v:Any, info:ValidationInfo)->Deployment:
         #     raise ValueError(f"Found more than 1 ({len(deps)}) Deployment for type '{typeField}' "\
         #                      f"from  possible {[deployment.type for deployment in director.deployments]}")
 
-
     @staticmethod
-    def get_detections_from_filenames(detection_filenames:set[str], all_detections:list[Detection_Abstract])->list[Detection_Abstract]:
+    def get_detections_from_filenames(
+        detection_filenames: set[str],
+        all_detections: list[Detection_Abstract]
+    ) -> list[Detection_Abstract]:
         detection_filenames = set(str(pathlib.Path(filename).absolute()) for filename in detection_filenames)
-        detection_dict = SecurityContentObject.create_filename_to_content_dict(all_detections)
+        detection_dict = SecurityContentObject.create_filename_to_content_dict(all_detections)      # type: ignore
 
         try:
+            # TODO (cmcginley): incompatible return type
             return [detection_dict[detection_filename] for detection_filename in detection_filenames]
```

Comment:
We are returning a list of `SecurityContentObject_Abstract`; we should be returning `Detection_Abstract`; perhaps `Detection_Abstract` should have an overwridden version of `create_filename_to_content_dict` to accomodate this?

Reply:
This function is actually not used anymore, so it has been removed.